clang/lib/Lex/HeaderSearch.cpp

```diff
@@ -1441,6 +1441,10 @@ void HeaderSearch::MarkFileModuleHeader(FileEntryRef FE,
   HFI.isCompilingModuleHeader |= isCompilingModuleHeader;
 }
 
+void HeaderSearch::EnteredTextualFile(FileEntryRef File) {
+  getFileInfo(File).isCompilingModuleHeader = true;
```

Comment:
The documentation for `isCompilingModuleHeader` says:
```
  /// Whether this header is part of the module that we are building, even if it
  /// doesn't build with the module. i.e. this will include `excluded` and
  /// `textual` headers as well as normal headers.
```
To me this means this should set only for headers that semantically belong to the current module, not headers that happen to be compiled with it due to being textual.

Reply:
Oops, I meant to alter that description in this patch, it seems that change got lost.

At this point, `isCompilingModuleHeader` is used only for this "is affecting" check, and for a heuristic related to translating `#include` into `#import`. The latter only cares about modular headers, so it seems reasonable to change the semantics for textual headers to something more useful. (But I do need to be careful we're only setting this for textual modular headers).